jax/_src/linear_util.py

```diff
@@ -271,6 +274,23 @@ def valid_size(d) -> bool:
   assert all(provided)
 
 
+class TracingDebugInfo(NamedTuple):
```

Comment:
This moved from api_util.py to here because we now treat it as a kind of annotation on lu.WrappedFun instances, hence we need it in this upstream file.